standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -5394,6 +5410,476 @@ public void countOpenTxns() throws MetaException {
+            + "(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)";
+
+    try {
+      preparedStatement = sqlGenerator.prepareStmtWithParameters(dbConn, insert, null);
+      for (Map.Entry entry : partitionInfoMap.entrySet()) {
+        ColumnStatistics colStats = (ColumnStatistics) entry.getValue();
+        PartitionInfo partitionInfo = (PartitionInfo)entry.getKey();
+        ColumnStatisticsDesc statsDesc = colStats.getStatsDesc();
+        long partId = partitionInfo.partitionId;
+
+        for (ColumnStatisticsObj statisticsObj : colStats.getStatsObj()) {
+          MPartitionColumnStatistics mPartitionColumnStatistics = StatObjectConverter.
+                  convertToMPartitionColumnStatistics(null, statsDesc, statisticsObj, colStats.getEngine());
+
+          preparedStatement.setLong(1, maxCsId);
+          preparedStatement.setString(2, mPartitionColumnStatistics.getCatName());
+          preparedStatement.setString(3, mPartitionColumnStatistics.getDbName());
+          preparedStatement.setString(4, mPartitionColumnStatistics.getTableName());
+          preparedStatement.setString(5, mPartitionColumnStatistics.getPartitionName());
+          preparedStatement.setString(6, mPartitionColumnStatistics.getColName());
+          preparedStatement.setString(7, mPartitionColumnStatistics.getColType());
+          preparedStatement.setLong(8, partId);
+          preparedStatement.setObject(9, mPartitionColumnStatistics.getLongLowValue());
+          preparedStatement.setObject(10, mPartitionColumnStatistics.getLongHighValue());
+          preparedStatement.setObject(11, mPartitionColumnStatistics.getDoubleHighValue());
+          preparedStatement.setObject(12, mPartitionColumnStatistics.getDoubleLowValue());
+          preparedStatement.setString(13, mPartitionColumnStatistics.getDecimalLowValue());
+          preparedStatement.setString(14, mPartitionColumnStatistics.getDecimalHighValue());
+          preparedStatement.setObject(15, mPartitionColumnStatistics.getNumNulls());
+          preparedStatement.setObject(16, mPartitionColumnStatistics.getNumDVs());
+          preparedStatement.setObject(17, mPartitionColumnStatistics.getBitVector());
+          preparedStatement.setObject(18, mPartitionColumnStatistics.getAvgColLen());
+          preparedStatement.setObject(19, mPartitionColumnStatistics.getMaxColLen());
+          preparedStatement.setObject(20, mPartitionColumnStatistics.getNumTrues());
+          preparedStatement.setObject(21, mPartitionColumnStatistics.getNumFalses());
+          preparedStatement.setLong(22, mPartitionColumnStatistics.getLastAnalyzed());
+          preparedStatement.setString(23, mPartitionColumnStatistics.getEngine());
+
+          maxCsId++;
+          numRows++;
+          preparedStatement.addBatch();
+          if (numRows == maxNumRows) {
+            preparedStatement.executeBatch();
+            numRows = 0;
+          }
+        }
+      }
+
+      if (numRows != 0) {
```

Comment:
maybe better "> 0" ?

Reply:
in this case it does not matter i think.